skimage/color/colorconv.py

```diff
@@ -156,6 +156,75 @@ def _prepare_colorarray(arr):
     return dtype.img_as_float(arr)
 
 
+def _prepare_rgba_array(arr):
+    """Check the shape of the array to be RGBA and convert it to
+    floating point representation.
+
+    """
+    arr = np.asanyarray(arr)
+
+    if arr.ndim not in [3, 4] or arr.shape[-1] != 4:
+        msg = ("the input array must have a shape == (.., ..,[ ..,] 4)), " +
+               "got (" + (", ".join(map(str, arr.shape))) + ")")
```

Comment:
Why do you need this? Would be much clearer and simpler just to feed tuple to `format`.